engine/src/main/java/org/terasology/rendering/dag/nodes/PrePostCompositeNode.java

```diff
@@ -58,23 +66,22 @@ public void initialise() {
     public void process() {
         PerformanceMonitor.startActivity("rendering/prePostComposite");
         prePostComposite.enable();
-        sceneOpaque = frameBuffersManager.getFBO("sceneOpaque");
-        sceneOpaquePingPong = frameBuffersManager.getFBO("sceneOpaquePingPong");
-        sceneReflectiveRefractive = frameBuffersManager.getFBO("sceneReflectiveRefractive");
+        sceneReflectiveRefractive = dynamicFBOsManager.get(REFLECTIVE_REFRACTIVE_URN);
 
         // TODO: verify if there should be bound textures here.
-        sceneOpaquePingPong.bind();
+        WRITE_ONLY_GBUFFER.bind();
 
-        setViewportToSizeOf(sceneOpaquePingPong);
+        setViewportToSizeOf(WRITE_ONLY_GBUFFER);
         glClear(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT); // TODO: verify this is necessary
 
         renderFullscreenQuad();
 
         bindDisplay();     // TODO: verify this is necessary
-        setViewportToSizeOf(sceneOpaque);    // TODO: verify this is necessary
+        setViewportToSizeOf(READ_ONLY_GBUFFER); // TODO: verify this is necessary
+
+        dynamicFBOsManager.swapReadWriteBuffers();
 
-        frameBuffersManager.swapSceneOpaqueFBOs();
-        sceneOpaque.attachDepthBufferTo(sceneReflectiveRefractive);
+        READ_ONLY_GBUFFER.attachDepthBufferTo(sceneReflectiveRefractive);
```

Comment:
Let's move this line above the swap and let's change it to WRITE_ONLY_GBUFFER, so that it's clearer that the data in it has just been written and gets available in the sceneReflectiveRefractive FBO.